client/core/core.go

```diff
@@ -1703,6 +1703,13 @@ var waiterID uint64
 func (c *Core) wait(assetID uint32, trigger func() (bool, error), action func(error)) {
 	c.waiterMtx.Lock()
 	defer c.waiterMtx.Unlock()
+	// TODO: While under lock, check the trigger before adding the waiter in
+	// case it is already satisfied.
+	// if ok, _ := trigger(); ok {
+	// 	log.Debugf("Got it on the first try. No waiter needed.")
+	// 	action(nil)
+	// 	return
+	// }
```

Comment:
As of now, `wait` is only used by `verifyRegistrationFee`, which in turn is only used by `Register`, where there will be zero confirmations, and `reFee`, which does it's own confirmation check before callling `wait`. You could move that confirmation check here, but unless we're expecting zero-confirmation registrations to be a thing, it will almost certainly result in an unnecessary confirmation check on the `Register` path.